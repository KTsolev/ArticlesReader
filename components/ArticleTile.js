
import React from "react";
import { View, Text, Image } from "react-native";
import { Icon } from 'react-native-elements';
import ProgressiveImage from './ProgresiveImage';

class ArticleTile extends React.Component {
    render() {
      const { article } = this.props;
      return <View style={{ flex: 1, flexDirection: 'row', padding: 10, alignItems: 'center', justifyContent: 'center'}}>
            <ProgressiveImage
              style={{ width: 65, height: 65, paddingRight: 20, borderRadius: 50}}
              thumbnailSource={{ uri: `https://images.pexels.com/photos/671557/pexels-photo-671557.jpeg?w=50&buster=${Math.random()}` }}
              source={{uri: article.media[0]['media-metadata'][0].url}}
              resizeMode="cover" />
            <View style={{ padding: 2, width: '70%', flexDirection: 'column', padding: 10}}>
                <View style={{ padding: 2, width: '100%', flexDirection: 'column'}}>
                    <Text numberOfLines={2} style={{ fontSize: 14, paddingTop: 5, paddingBottom: 5, color: '#383838', fontWeight: 'bold' }}>{article.title}</Text>
                    <Text numberOfLines={2} style={{ fontSize: 12, paddingTop: 5, paddingBottom: 5, color: '#5b5e61' }}>{article.abstract}</Text>
                </View>
                <View style={{ width: '100%', flexDirection: 'row'}}>
                    <View style={{ padding: 2, width: '50%', alingItems: 'flex-start', justifyContent: 'center', wordWrap: 'wrap'}}>
                        <Text numberOfLines={2} style={{ fontSize: 10, color: '#5b5e61' }}>{article.byline}</Text>
                    </View>
                    <View style={{ padding: 2, paddingLeft: 20, width: '50%', flexDirection: 'row', alingItems: 'flex-end', justifyContent: 'flex-end'}}>
                        <View style={{ width: '30%', paddingTop: '10%', paddingBottom: '10%'}} >
                          <Icon name='calendar' size={14} type='font-awesome' color='#5b5e61' />
                        </View>  
                        <Text style={{ width: '70%', fontSize: 10, color: '#5b5e61', paddingTop: '10%', paddingBottom: '10%'}}>{article.published_date}</Text>
                    </View>
                </View>
            </View>
            <View style={{ paddingLeft: 20 }}>
              <Icon  name='chevron-right' size={16} type='font-awesome' color='#5b5e61' />
            </View>
        </View>;
    }
  }
 
export default ArticleTile;