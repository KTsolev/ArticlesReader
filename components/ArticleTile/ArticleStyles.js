import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
   articleContainer: { flex: 1, flexDirection: 'row', padding: 10, alignItems: 'center',
      justifyContent: 'center'},
   articleImageHolder: { width: 65, height: 65, paddingRight: 20, borderRadius: 50},
   articleHaederContainer: { padding: 2, width: '100%', flexDirection: 'column'},
   articleHeaderTitle: { fontSize: 14, paddingTop: 5, paddingBottom: 5, 
      color: '#383838', fontWeight: 'bold' },
   articleHeaderSubTitle: { fontSize: 12, paddingTop: 5, paddingBottom: 5, color: '#5b5e61' },
   articleTileFooter: { width: '100%', flexDirection: 'row'},
   articleAuthorHolder: { padding: 2, width: '50%', alignItems: 'flex-start',
      justifyContent: 'center', flexWrap: 'wrap'},
   articleAuthorText: { fontSize: 10, color: '#5b5e61' },
   articleFooterDateHolder: { padding: 2, paddingLeft: 20, width: '50%',
      flexDirection: 'row', alignContent: 'center', justifyContent: 'flex-end'},
   articleFooterIconHolder: { width: '30%', paddingTop: '10%', paddingBottom: '10%',
      alignContent: 'center'},
   articleTileMain: {width: '70%', fontSize: 10, color: '#5b5e61', paddingLeft: 5,
      paddingTop: '10%', paddingBottom: '10%'},
   articleDateText: {width: '70%', fontSize: 10, color: '#5b5e61', paddingTop: '10%',
      paddingBottom: '10%'},
   articleAsside: { paddingLeft: 20 }
});