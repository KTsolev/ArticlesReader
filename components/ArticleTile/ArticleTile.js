
import React from 'react';
import { View, Text } from 'react-native';
import { Icon } from 'react-native-elements';
import ProgressiveImage from '../ProgresiveImage/ProgresiveImage';
import { styles } from './ArticleStyles';

class ArticleTile extends React.Component {
    render() {
      const { article } = this.props;
      const {
        articleContainer,
        articleImageHolder,
        articleTileMain,
        articleHaederContainer,
        articleHeaderTitle,
        articleHeaderSubTitle,
        articleTileFooter,
        articleAuthorHolder,
        articleAuthorText,
        articleFooterDateHolder,
        articleFooterIconHolder,
        articleDateText,
        articleAsside
      } = styles;
     
     return <View style={articleContainer}>
            <ProgressiveImage
              style={articleImageHolder}
              thumbnailSource={require('../../images/dummy-image.png')}
              source={{uri: article.media[0]['media-metadata'][0].url}}
              resizeMode="contain" />
            <View style={articleTileMain}>
                <View style={articleHaederContainer}>
                    <Text numberOfLines={2} style={articleHeaderTitle}>{article.title}</Text>
                    <Text numberOfLines={2} style={articleHeaderSubTitle}>{article.abstract}</Text>
                </View>
                <View style={articleTileFooter}>
                    <View style={articleAuthorHolder}>
                        <Text numberOfLines={2} style={articleAuthorText}>{article.byline}</Text>
                    </View>
                    <View style={articleFooterDateHolder}>
                        <View style={articleFooterIconHolder} >
                          <Icon name='calendar' size={14} type='font-awesome' color='#5b5e61' />
                        </View>  
                        <Text style={articleDateText}>{article.published_date}</Text>
                    </View>
                </View>
            </View>
            <View style={articleAsside}>
              <Icon  name='chevron-right' size={16} type='font-awesome' color='#5b5e61' />
            </View>
        </View>;
    }
  }
 
export default ArticleTile;