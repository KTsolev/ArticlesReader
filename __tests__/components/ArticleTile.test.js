import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { render } from 'react-native-testing-library';
import ArticleTile from '../../components/ArticleTile/ArticleTile';


const createTestProps = (props) => ({
    article: {
        title: 'Miners Kill Indigenous Leader in Brazil During Invasion of Protected Land',
        abstract: 'Land invasions in indigenous territories are on the rise across Brazil. Indigenous leaders say they regularly come under threat by miners, loggers and farmers',
        published_date: '2019-07-27',
        url: 'https://www.nytimes.com/2019/07/27/world/americas/brazil-miners-amapa.html',
        byline: 'By ERNESTO LONDOÃ‘O',
        media: [
            {
                'media-metadata': [
                    {
                        'url': 'https://static01.nyt.com/images/2019/07/27/world/Brazil-Indigenous/Brazil-Indigenous-square320.jpg',
                        'format': 'square320',
                        'height': 320,
                        'width': 320
                    },
                ]
            }
        ]
    },
    ...props
    });

describe('<ArticleTile />', () => {
    let props;
    
    beforeEach(() => {
        props = createTestProps({});
    });
    
    test('renders the component', () => {
        const wrapper = shallow(<ArticleTile {...props} />);
        const component = wrapper.dive();
        expect(toJson(component)).toMatchSnapshot();
    });
    test('renders the component check if props are applied', () => {
        const { queryByText } = render(<ArticleTile {...props} />);
        expect(queryByText(props.article.title)).not.toBeNull();
        expect(queryByText(props.article.abstract)).not.toBeNull();
        expect(queryByText(props.article.published_date)).not.toBeNull();
        expect(queryByText(props.article.byline)).not.toBeNull();
    });
});

