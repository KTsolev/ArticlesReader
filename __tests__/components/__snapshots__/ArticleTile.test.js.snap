// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleTile /> renders the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <ProgressiveImage
    resizeMode="contain"
    source={
      Object {
        "uri": "https://static01.nyt.com/images/2019/07/27/world/Brazil-Indigenous/Brazil-Indigenous-square320.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 50,
        "height": 65,
        "paddingRight": 20,
        "width": 65,
      }
    }
    thumbnailSource={
      Object {
        "testUri": "../../../images/dummy-image.png",
      }
    }
  />
  <View
    style={
      Object {
        "color": "#5b5e61",
        "fontSize": 10,
        "paddingBottom": "10%",
        "paddingLeft": 5,
        "paddingTop": "10%",
        "width": "70%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "padding": 2,
          "width": "100%",
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#383838",
            "fontSize": 14,
            "fontWeight": "bold",
            "paddingBottom": 5,
            "paddingTop": 5,
          }
        }
      >
        Miners Kill Indigenous Leader in Brazil During Invasion of Protected Land
      </Text>
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#5b5e61",
            "fontSize": 12,
            "paddingBottom": 5,
            "paddingTop": 5,
          }
        }
      >
        Land invasions in indigenous territories are on the rise across Brazil. Indigenous leaders say they regularly come under threat by miners, loggers and farmers
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "padding": 2,
            "width": "50%",
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Object {
              "color": "#5b5e61",
              "fontSize": 10,
            }
          }
        >
          By ERNESTO LONDOÃ‘O
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "padding": 2,
            "paddingLeft": 20,
            "width": "50%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "paddingBottom": "10%",
              "paddingTop": "10%",
              "width": "30%",
            }
          }
        >
          <Themed.Icon
            color="#5b5e61"
            name="calendar"
            size={14}
            type="font-awesome"
          />
        </View>
        <Text
          style={
            Object {
              "color": "#5b5e61",
              "fontSize": 10,
              "paddingBottom": "10%",
              "paddingTop": "10%",
              "width": "70%",
            }
          }
        >
          2019-07-27
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingLeft": 20,
      }
    }
  >
    <Themed.Icon
      color="#5b5e61"
      name="chevron-right"
      size={16}
      type="font-awesome"
    />
  </View>
</View>
`;
